### ISSUE TITLE
why go?

### ISSUE THREAD

**Issue by chpio** [fc00/go-fc00#1]
Also it's not about *[rewriting](https://unhandledexpression.com/2017/07/10/why-you-should-actually-rewrite-it-in-rust/)*, fc00 is a new project. Its is a mesh networking protocol, so performance should be a top priority, do you write kernels in gc langs? So why do you want to write a *performance critical* software in a GC lang like go?

PS:  i thought while i was writing this that fc00 will replace the original c implementation (https://github.com/cjdelisle/cjdns).

**Comment by ghost** [comment]
Why haven't you written it in rust?

**Comment by chpio** [comment]
hmm, my actual question was about building a maybe more valuable solution (rust can run in more environments due to the minimalistic runtime) and combining the effort and not splitting it up by maintaining multiple implementations.

**Comment by chpio** [comment]
https://github.com/ipfs/go-ipfs/issues/2467 ipfs is "more high level", so it's "okay" if it is not that "minimalistic". But lowlevel stuff like cjdns/fc00 should be able to start without allocating hundreds of MB (imho).

**Comment by ghost** [comment]
It wouldn't make any sense to write go-fc00 in Rust, that would just be too confusing.

Maybe you want https://github.com/Randati/cjdrs.

**Comment by ghost** [comment]
I wanna integrate it into go-ipfs so rust is not an option right now. @chpio I'd be happy if _you_ wrote it in Rust ;)

**Comment by ansuz** [comment]
## My hobby:

> Compiling lists of instances of rust aficionados asking other people to write or rewrite large projects in Rust instead of the language of their choice

**Comment by cjdelisle** [comment]
:100:

**Comment by chpio** [comment]
ok, didn't thought a simple question would attract that much angry trolls. Anyway, as long there is an updated low level implementation (currently the original cjdns) i will be happy.

**Comment by ansuz** [comment]
I actually think Rust is a really cool language. I've just noticed that for whatever reason this is a very prevalent meme in the Rust community.

I am, in fact, compiling such a list to support this assertion, and it will be featured on my blag very soon. It will feature an issue filed on @cjdelisle's cjdns repo, so this is something of a duplicate.

That being said, we take issues seriously. @lgierth's requirements for compatibility with go-ipfs are legitimate, and @ehmry's suggestion to check out https://github.com/Randati/cjdrs seems quite helpful.

I've looked into this problem before, and I suggest checking out https://github.com/dnaq/sodiumoxide
